src/System.Net.Primitives/tests/FunctionalTests/CookieTest.cs

```diff
@@ -240,13 +240,13 @@ public static void Value_GetSet_Success()
             Assert.Equal(string.Empty, c.Value);
         }
         
-        // NOTE: This test is expected to fail on the full desktop.
-        // The behavior this tests was only recently added in #8206.
         [Fact]
         public static void Value_PassNullToCtor_GetReturnsEmptyString()
         {
             var cookie = new Cookie("SomeName", null);
-            Assert.Equal(string.Empty, cookie.Value);
+            // cookie.Value will return empty string on netcore and null on full framework.
+            // The behavior this tests was only recently added in #8206.
+            Assert.True(string.IsNullOrEmpty(cookie.Value));
```

Comment:
Can this be made more discreet, i.e. do a runtime check and if net46 then test for null but if netcore test for empty string?  That would help pin the expected behavior more accurately.